codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientHandshaker00.java

```diff
@@ -201,20 +202,20 @@ protected void verify(FullHttpResponse response) {
     }
 
     private static String insertRandomCharacters(String key) {
-        int count = WebSocketUtil.randomNumber(1, 12);
+        int count = ThreadLocalRandom.current().nextInt(1, 13);
 
         char[] randomChars = new char[count];
         int randCount = 0;
         while (randCount < count) {
-            int rand = (int) (Math.random() * 0x7e + 0x21);
+            int rand = ThreadLocalRandom.current().nextInt(0x21, 0x7e + 0x21);
             if (0x21 < rand && rand < 0x2f || 0x3a < rand && rand < 0x7e) {
```

Comment:
@normanmaurer this check also doesn't seem correct to me.
Random is generated in range `0x21 .. 0x7e + 0x21`, however, values above` 0x7e` are skipped...